src/node/mini_miner.cpp

```diff
@@ -189,11 +189,7 @@ struct AncestorFeerateComparator
             const int64_t ancestor_size{e.GetSizeWithAncestors()};
             const CAmount tx_fee{e.GetModifiedFee()};
             const int64_t tx_size{e.GetTxSize()};
-            // Comparing ancestor feerate with individual feerate:
-            //     ancestor_fee / ancestor_size <= tx_fee / tx_size
-            // Avoid division and possible loss of precision by
-            // multiplying both sides by the sizes:
-            return ancestor_fee * tx_size < tx_fee * ancestor_size ?
+            return CFeeRate(ancestor_fee, ancestor_size) <= CFeeRate(tx_fee, tx_size) ?
```

Comment:
instead of b1b28a21e185dfbc0c75f92ac55401a9ba8a5b08, I'd recommend rebasing on #30412
I was able to reproduce the ub crash and verify that 6254c253e995e77196b853784cfd0f42619b324d fixes it.

Reply:
thanks I will rebase on it.

Reply:
done